### DISCUSSION TITLE
How to make ComboBox scollable?

### DISCUSSION THREAD

**Discussion by whatakey** [slint-ui/slint#2657] [Q&A] [answered]
Hi

I have a ComboBox with many items in it, I want it to display only 10 and show scroll bar used to reveal more items, any code example of that would be helpful.

Thank you.

**Comment by ogoffart** [comment, marked as answer]
The default ComboBox don't do that.
But you can re-implement one that does that.
For example, i took the one from the printer demo and added a Flickable in it: [SlintPad Link](https://slint-ui.com/releases/1.0.2/editor/?snippet=import+%7B+Button%2C+VerticalBox%2C+StyleMetrics+%2C+ScrollView%2C+HorizontalBox%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+ComboBox+%7B%0A++++in-out+property%3Cstring%3E+value%3B%0A++++in+property%3C%5Bstring%5D%3E+choices%3B%0A++++Rectangle+%7B%0A++++++++border-radius%3A+3px%3B%0A++++++++border-width%3A+2px%3B%0A++++++++border-color%3A+gray%3B%0A++++%7D%0A++++height%3A+32px%3B%0A++++preferred-width%3A+100%25%3B%0A++++HorizontalBox+%7B+%0A++++++++label+%3A%3D+Text+%7B%0A++++++++++++vertical-alignment%3A+center%3B%0A++++++++++++horizontal-alignment%3A+left%3B%0A++++++++++++text+%3C%3D%3E+root.value%3B%0A++++++++++++color%3A+black%3B%0A++++++++++++horizontal-stretch%3A+1%3B%0A++++++++%7D%0A++++++++Text+%7B%0A++++++++++++text%3A+%22%E2%96%BC%22%3B%0A++++++++%7D%0A++++%7D%0A%0A++++TouchArea+%7B%0A++++++++width%3A+100%25%3B%0A++++++++height%3A+100%25%3B%0A++++++++clicked+%3D%3E+%7B+popup.show%28%29%3B+%7D%0A++++%7D%0A%0A++++popup+%3A%3D+PopupWindow+%7B%0A++++++++x%3A0%3B%0A++++++++y%3A+root.height%3B%0A++++++++width%3A+root.width%3B%0A++++++++height%3A+min%28inner.preferred-height%2C+120px%29%3B%0A++++++++Rectangle+%7B%0A++++++++++++background%3A+StyleMetrics.window-background%3B%0A++++++++++++border-radius%3A+3px%3B%0A++++++++++++border-width%3A+2px%3B%0A++++++++++++border-color%3A+gray%3B%0A++++++++%7D%0A++++++++ScrollView+%7B%0A++++++++++++inner+%3A%3D+VerticalLayout+%7B%0A++++++++++++++++spacing%3A+6px%3B%0A++++++++++++++++padding%3A+3px%3B%0A++++++++++++++++for+value%5Bidx%5D+in+root.choices%3A+Rectangle+%7B%0A++++++++++++++++++++border-radius%3A+3px%3B%0A++++++++++++++++++++background%3A+item-area.has-hover+%3F+%23eeee+%3A+%230000%3B%0A++++++++++++++++++++HorizontalLayout+%7B%0A++++++++++++++++++++++++Text+%7B%0A++++++++++++++++++++++++++++text%3A+value%3B%0A++++++++++++++++++++++++%7D%0A++++++++++++++++++++%7D%0A++++++++++++++++++++item-area+%3A%3D+TouchArea+%7B%0A++++++++++++++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++++++++++++++root.value+%3D+value%3B%0A++++++++++++++++++++++++%7D%0A++++++++++++++++++++%7D%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++ComboBox+%7B%0A++++++++++++value%3A+%22hello%22%3B%0A++++++++++++choices%3A+%5B%22lorem%22%2C+%22ipsum%22%2C+%22dolor%22%2C+%22sit%22%2C+%22amet%22%2C+%22hello%22%2C+%22world%22%2C+%22lorem%22%2C+%22ipsum%22%2C+%22dolor%22%2C+%22sit%22%2C+%22amet%22%2C+%22hello%22%2C+%22world%22%5D%3B%0A++++++++%7D%0A++++++++Text+%7B%0A++++++++++++text%3A+%22Hello+World%21%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++%7D%0A++++++++Image+%7B%0A++++++++++++source%3A+%40image-url%28%22https%3A%2F%2Fslint-ui.com%2Flogo%2Fslint-logo-full-light.svg%22%29%3B%0A++++++++++++height%3A+100px%3B%0A++++++++%7D%0A++++++++HorizontalLayout+%7B+alignment%3A+center%3B+Button+%7B+text%3A+%22OK%21%22%3B+%7D+%7D%0A++++%7D%0A%7D%0A)

**Comment by whatakey** [comment]
Thank you for your example and sorry for the trouble, I thought it was something simple!
